## joint_names
joint_names: [f_caster_rotation_joint, f_caster_r_wheel_joint, br_caster_rotation_joint, br_caster_r_wheel_joint, bl_caster_rotation_joint, bl_caster_r_wheel_joint]

## joint_state_controller
joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

f_caster_rotation_joint_position_controller:
  type: position_controllers/JointPositionController
  joint: f_caster_rotation_joint
  required_drive_mode: 1
br_caster_rotation_joint_position_controller:
  type: position_controllers/JointPositionController
  joint: br_caster_rotation_joint
  required_drive_mode: 1
bl_caster_rotation_joint_position_controller:
  type: position_controllers/JointPositionController
  joint: bl_caster_rotation_joint
  required_drive_mode: 1

## twist_controller
twist_controller:
  type: cob_omni_drive_controller/WheelController
  required_drive_mode: 3
  #max_rot_velocity: 1.0
  #max_trans_velocity: 1.0
  timeout: 0.5

  defaults: # default settings for all wheels, can per overwritten per wheel
    steer_ctrl: # Parameters for tuning Impedance-Controller
      spring: 15.0
      damp: 2.75
      virt_mass: 0.1
      d_phi_max: 10.0
      dd_phi_max: 40.0

  wheels: &wheels # start front left, counter-clockwise for base_drive_chain compatibility
    - steer: f_caster_rotation_joint
      drive: f_caster_r_wheel_joint
    - steer: bl_caster_rotation_joint
      drive: bl_caster_r_wheel_joint
    - steer: br_caster_rotation_joint
      drive: br_caster_r_wheel_joint

## odometry_controller
odometry_controller:
  type: cob_omni_drive_controller/OdometryController
  publish_rate: 50
  frame_id: "/odom_combined"
  child_frame_id: "/base_footprint"
  cov_pose: 0.1
  cov_twist: 0.1
  wheels: *wheels
